modules/vcf2maf/1.3/vcf2maf.smk

```diff
@@ -222,88 +383,169 @@ rule _vcf2maf_crossmap:
-        maf_open["Chromosome"] = maf_open["Chromosome"].astype(str).str.replace('chr', '')
-        if params.dest_chr:  # Will evaluate to True if the destination genome is chr-prefixed
-            # Add chr prefix
-            maf_open['Chromosome'] = 'chr' + maf_open['Chromosome'].astype(str)
-
-        maf_open.to_csv(output.maf, sep="\t", index=False)
-
-rule _vcf2maf_output_maf_projection:
-    input:
-        maf = str(rules._vcf2maf_normalize_prefix.output.maf),
-    output:
-        maf = CFG["dirs"]["outputs"] + "{seq_type}--{target_build}--projection/{tumour_id}--{normal_id}--{pair_status}_{base_name}.maf"
-    wildcard_constraints:
-        target_build = "|".join(CFG["options"]["target_builds"]), 
-        base_name = CFG["vcf_base_name"]
-    run:
-        op.relative_symlink(input.maf, output.maf)
+        if params.todo == "symlink": 
+            op.relative_symlink(input.maf, output.maf)
+        else: 
+            maf_open = pd.read_csv(input.maf[0], sep = "\t")
+            print("Normalizing " + input.maf[0])
+            # To handle CrossMap weirdness, remove all chr-prefixes and add them back later
+            maf_open["Chromosome"] = maf_open["Chromosome"].astype(str).str.replace('chr', '')
+            if params.dest_chr:  # Will evaluate to True if the destination genome is chr-prefixed
+                # Add chr prefix
+                maf_open['Chromosome'] = 'chr' + maf_open['Chromosome'].astype(str)
+            print("Writing to " + output.maf)
+            maf_open.to_csv(output.maf, sep="\t", index=False)
+        
 
 rule _vcf2maf_output_maf:
     input:
         maf = str(rules._vcf2maf_normalize_prefix.output.maf),
     output:
-        maf = CFG["dirs"]["outputs"] + "{seq_type}--projection/{tumour_id}--{normal_id}--{pair_status}_{base_name}.{target_build}.maf"
+        maf = CFG["dirs"]["outputs"] + "{filter}/{maf}/{seq_type}--projection/{tumour_id}--{normal_id}--{pair_status}.{base_name}.{target_build}.maf"
     wildcard_constraints:
         target_build = "|".join(CFG["options"]["target_builds"]), 
         base_name = CFG["vcf_base_name"]
     run:
-        op.relative_symlink(input.maf, output.maf)
+        op.relative_symlink(input.maf, output.maf, in_module = True)
+
+print(CFG["options"]["filter"])
 
 # Generates the target sentinels for each run, which generate the symlinks
 rule _vcf2maf_all:
     input:
         expand(
-            expand(str(rules._vcf2maf_output_maf.output.maf), zip,
+            expand(
+                str(rules._vcf2maf_output_original.output.maf), 
+                zip, 
+                seq_type = CFG["runs"]["tumour_seq_type"], 
+                genome_build = CFG["runs"]["tumour_genome_build"], 
+                tumour_id = CFG["runs"]["tumour_sample_id"], 
+                normal_id = CFG["runs"]["normal_sample_id"], 
+                pair_status = CFG["runs"]["pair_status"], 
+                allow_missing = True
+            ),
+            maf = "maf", 
```

Comment:
here and below `maf = "maf"` seems to be one element. I wonder how this works, especially where there is expand on at least two items: `filter = CFG["options"]["filter"]`  is set to `filter: ["raw", "deblacklisted"]` so I would think the maf  needs to be something like `maf = ["maf"]*length(CFG["options"]["filter"])` to generate 2 files, in this case `raw/maf` and `deblacklisted/maf`. Otherwise if the length is 1, I expect it to produce only the first instance after expansion, so only `raw/maf`. Asking because this might help me to wrap my head around the CNV merging 😄

Reply:
I think this is because it's set up to be run twice with the config specifically set to do this for either the raw or deblacklisting phase (separated on purpose for blacklist creation)